pages/api/models.ts

```diff
@@ -4,43 +4,37 @@ export const config = {
   runtime: "edge"
 };
 
+export async function getModels(key: string): Promise<OpenAIModel[]> {
+  const response = await fetch('https://api.openai.com/v1/models', {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${key ? key : process.env.OPENAI_API_KEY}`
+    }
+  });
+
+  const json = await response.json();
+
+  const models: OpenAIModel[] = json.data
+  .map((model: any) => {
+    for (const [ key, value ] of Object.entries(OpenAIModelID)) {
+      if (value === model.id) {
+        return {
+          id: model.id,
+          name: OpenAIModels[value].name
+        };
+      }
+    }
+  })
+  .filter(Boolean);
+  return models;
+}
```

Comment:
Use `Promise.all()` to make multiple API requests in parallel. This can improve the performance of your code, especially if you need to make multiple requests to different endpoints.

Reply:
Whilst we can do this, I didnt want to change the actual logic / implementation of the server code in this PR. It is out of the scope of it.

I think @mckaywrigley is planning some work for refactoring / optimisations at a later point so maybe we could pick it up then?